### PR TITLE
Adding quadprog interfaces

### PR DESCRIPTION
**Pull request by RenanOD**

### CODE REVIEW

File: src/quadprog.jl [hunk 1]
```diff
@@ -1,13 +1,74 @@
 export quadprog
 
+const MatrixOrOperator = Union{AbstractMatrix, AbstractLinearOperator}
```
Comment (abelsiqueira) [inline src/quadprog.jl]: This could be put inside `auxiliary.jl`, and removed from here and from `linprog`.

File: src/quadprog.jl [hunk 2]
```diff
@@ -1,13 +1,74 @@
...
+  nvar = length(g)
+  @assert size(Q, 1) == size(Q, 2) == nvar
+  @assert length(lvar) == length(uvar) == nvar
+  @assert size(Aineq, 2) == size(Aeq, 2) == nvar
+  @assert size(Aineq, 1) == length(blow) && size(Aineq, 1) == length(bupp)
+  @assert size(Aeq, 1) == length(beq)
   f(x) = dot(x, Q*x)/2 + dot(g,x)
-  con(x) = A*x
-  nlp = ADNLPModel(f, zeros(length(g)), c = con, ucon = b)
+  con = x -> [Aineq * x; Aeq * x]
```
Comment (abelsiqueira) [inline src/quadprog.jl]: Write as `con(x) = ...`

**Comment by RenanOD** [comment]
It's ready for review, should be back to 100% coverage.

**Comment by RenanOD** [comment]
> For quadratic programming, it also makes sense to have an unconstrained problem: `quadprog(Q, g)`, and only equality constraints, i.e., no bounds nor inequalities. I'm not sure the best interface for only equalities that won't clash with only inequalities. In MatLab, one would use `quadprog(Q, g, [], [], Aeq, beq)`, but I hope we can have a better solution.

The general quadprog function is `quadprog(Q, g; ...)`, the first test is unconstrained.
I'll be thinking about a better solution than `quadprog(Q, g, [], [], Aeq, beq)`.

**Comment by RenanOD** [comment]
Currently `quadprog(Q, g, Aeq = Aeq, beq = beq` is supported, wouldn't describing this in the help expression be enough?

**Comment by abelsiqueira** [comment]
I think that for now, we can leave like that.

**Comment by RenanOD** [comment]
Ready for review again.

**Comment by abelsiqueira** [comment]
Thank you

**Pull request merged**